testing/file_test/file_test_base.h

```diff
@@ -110,6 +110,19 @@ class FileTestBase {
   // Returns the name of the test (relative to the repo root).
   auto test_name() const -> llvm::StringRef { return test_name_; }
 
+  // Returns a bazel label that can be used to invoke this test.
+  virtual auto GetBazelLabel() -> std::string;
+
+  // Modes for GetBazelCommand.
+  enum class BazelMode : uint8_t {
+    Autoupdate,
+    Dump,
+    Test,
+  };
+
+  // Returns the requested bazel command string for the given execution mode.
+  virtual auto GetBazelCommand(BazelMode mode) -> std::string;
```

Comment:
Does this need to be `virtual`?

```suggestion
  auto GetBazelCommand(BazelMode mode) -> std::string;
```